src/Integration.Vsix/SolutionAnalysisRequester.cs

```diff
@@ -0,0 +1,106 @@
+//-----------------------------------------------------------------------
+// <copyright file="SolutionAnalysisRequester.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Options;
+using System;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+
+namespace SonarLint.VisualStudio.Integration.Vsix
+{
+    internal class SolutionAnalysisRequester
+    {
+        private readonly Workspace workspace;
+
+        private object optionService;
+        private Option<bool> option;
+        private MethodInfo optionServiceSetOptionsMethod;
+        private MethodInfo optionServiceGetOptionsMethod;
+
+        public SolutionAnalysisRequester(Workspace workspace)
+        {
+            if (workspace == null)
+            {
+                throw new ArgumentNullException(nameof(workspace));
+            }
+
+            this.workspace = workspace;
+            this.FindOptionService();
+            this.FindFullSolutionAnalysisOption();
+        }
+
+        private void FindOptionService()
+        {
+            var types = typeof(PerLanguageOption<int>).Assembly.GetTypes();
+            var interfaces = types.Where(t => t.IsInterface);
+
+            var optionServiceType = interfaces.FirstOrDefault(t => t.Name == "IOptionService");
+            if (optionServiceType == null)
+            {
+                return;
+            }
+
+            this.optionService = this.workspace.Services.GetType()
+                .GetMethod("GetService")
+                ?.MakeGenericMethod(optionServiceType)
+                ?.Invoke(workspace.Services, null);
+
+            this.optionServiceSetOptionsMethod = optionServiceType?.GetMethod("SetOptions");
+            this.optionServiceGetOptionsMethod = optionServiceType?.GetMethod("GetOptions");
+        }
+
+        private void FindFullSolutionAnalysisOption()
+        {
+            //there's no public type in the DLL, so we try finding it next to Microsoft.CodeAnalysis
+            var path = Path.Combine(
+                new FileInfo(typeof(Accessibility).Assembly.Location).Directory.FullName,
+                "Microsoft.CodeAnalysis.Features.dll");
+
+            if (!File.Exists(path))
+            {
+                return;
+            }
+
+            this.option = (Option<bool>)Assembly.LoadFile(path)
+                .GetTypes()
```

Comment:
Can we use GetType("RuntimeOptions") Do we really care about the other stuff ? in case it won't have that static field it will no op anyway.